### PR TITLE
Added LineIntersection class

### PR DESCRIPTION
**Pull request by maik-nack**

### CODE REVIEW

File: app/src/Elements/LineIntersection.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            case 'line':
+              return (obj as Line).intersectWithLine(element as Line);
+          }
+        };
+        break;
+      default:
+        return undefined;
+    }
+
+    for (let i = 1; i < objects.length; i++) {
```
Comment (leins275) [inline app/src/Elements/LineIntersection.tsx:85]: i = 1 (not 0) ??

**Review by leins275 — CHANGES_REQUESTED** [review]

File: app/src/Elements/LineIntersection.tsx (hunk 1 above)
Comment (maik-nack) [reply app/src/Elements/LineIntersection.tsx:85]: `getIntersection` is a function for a search of the intersection of `objects[0]` and another object. So, why we need to intersect `objects[0]` with `objects[0]`? In addition, the result of that intersection exists only when `objects[0]` is the point.

Comment (leins275) [reply app/src/Elements/LineIntersection.tsx:85]: Ok, understand

**Review by leins275 — APPROVED** [review]
Ok

**Pull request merged**